src/addons/addonmessage.cpp

```diff
@@ -224,3 +286,28 @@ qint64 AddonMessage::planDateRetranslationInternal(
 
   return 86400 + secsTo;
 }
+
+void AddonMessage::enable() {
+  Addon::enable();
+
+  maybePushNotification();
+}
+
+void AddonMessage::maybePushNotification() {
+  if (!m_conditionsParsed) {
```

Comment:
in theory, `::enable()` is called the first time _after_ maybeCreateConditionWatchers(): https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/addons/addon.cpp#L349-L352

If this happens before, it's a bug.

Reply:
Correct. Development was a bit rocky on his feature and this was added before the AddonMessage::State which makes this whole `m_conditionsParsed` unnecessary it seems like :thinking:  I did not realize that until reading this comment. I'll just say goodbye to that state.